smack-extensions/src/main/java/org/jivesoftware/smackx/softwareinfo/SoftwareInfoManager.java

```diff
@@ -0,0 +1,95 @@
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPException.XMPPErrorException;
+import org.jivesoftware.smack.parsing.SmackParsingException;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfo;
+import org.jivesoftware.smackx.softwareinfo.form.SoftwareInfoForm;
+import org.jivesoftware.smackx.xdata.packet.DataForm;
+
+import org.jxmpp.jid.EntityJid;
+import org.jxmpp.jid.Jid;
+
+/**
+* Entry point for Smack's API for XEP-0232: Software Information.
+* <br>
+* @see <a href="https://xmpp.org/extensions/xep-0232.html">
+*     XEP-0232 : Software Information.
+*     </a>
+*/
+public final class SoftwareInfoManager extends Manager {
+
+    private static final String FEATURE = "http://jabber.org/protocol/disco";
+    private static final Map<XMPPConnection, SoftwareInfoManager> INSTANCES = new WeakHashMap<>();
+    private final ServiceDiscoveryManager serviceDiscoveryManager;
+
+    public static synchronized SoftwareInfoManager getInstanceFor (XMPPConnection connection) throws IOException, XmlPullParserException, SmackParsingException {
+        SoftwareInfoManager manager = INSTANCES.get(connection);
+        if (manager == null) {
+            manager = new SoftwareInfoManager(connection);
+            INSTANCES.put(connection, manager);
+        }
+        return manager;
+    }
+
+    private SoftwareInfoManager(XMPPConnection connection) throws IOException, XmlPullParserException, SmackParsingException {
+        super(connection);
+        serviceDiscoveryManager = ServiceDiscoveryManager.getInstanceFor(connection);
+    }
+
+    /**
+     * Returns true if the feature is supported by the Jid.
+     * <br>
+     * @param jid Jid to be checked for support
+     * @return boolean
+     * @throws NoResponseException if there was no response from the remote entity
+     * @throws XMPPErrorException if there was an XMPP error returned
+     * @throws NotConnectedException if the XMPP connection is not connected
+     * @throws InterruptedException if the calling thread was interrupted
+     */
+    public boolean isSupported(Jid jid) throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return serviceDiscoveryManager.supportsFeatures(jid, FEATURE);
+    }
+
+    public void publishSoftwareInformationForm(SoftwareInfoForm softwareInfoForm) {
+        serviceDiscoveryManager.setExtendedInfo(softwareInfoForm.getDataFormToSend());
+    }
+
+    public SoftwareInfoForm fromJid(EntityJid jid) throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        DiscoverInfo discoverInfo = serviceDiscoveryManager.discoverInfo(jid);
+        DataForm dataForm = DataForm.from(discoverInfo);
```

Comment:
That disregards the aspect that there could be potentially multiple data forms in this disco#info response.